### DISCUSSION TITLE
download zip files windows error

### DISCUSSION THREAD

**Discussion by DigitalTrading** [openstore-ecommerce/OpenStore#283] [General]
Hi,
I have attached zip files documents to a product as a download. After the user makes a purchased they click the download link and the zip file downloads as expected. When the user tries to open the folder we are getting this windows error:

windows cannot open the folder
The compressed zip folder is invalid

Can you advise what could be causing this?
Regards
Roger

**Comment by DNNMonster** [comment]
Are you manually creating the zip file or does an app create those?  My first thought would be that something is wrong in the code that is creating the zip file.  If you created it manually then that is not likely the problem.

**Reply by DigitalTrading** [reply]
We create the zip initially and upload to the product. This is saved under a different file name something like (ffec3bcf-c2c3-4363-953d-09c10aa63957.zip). The download is created from the app and arrives back with the correct file name. This is the file which is corrupt. I have tested the zip file on the server which opens with no issues.

**Reply by DNNMonster** [reply]
I just tested zipping a yml file, uploaded to an Azure DNN VM and pulled back down..

I am not having any problems..  

Are you seeing this on just 1 environment or multiple?

**Reply by DigitalTrading** [reply]
The issue for me is on every one which are zip files. pdfs which are not zipped open fine.

**Reply by DNNMonster** [reply]
On the latest OS version?

**Reply by DNNMonster** [reply]
Well I know we are probably looking around here but I am not certain what it is..??  Maybe the context.Response?

You could put a break point in and see if anything stands out.


https://github.com/openstore-ecommerce/OpenStore/blob/4af26cc2c45da5a4c524526f3c7b2379f4be11a7/XmlConnector.ashx.cs#L304

**Comment by DNNMonster** [comment]
I'll test a remote server.  Testing on localhost I did not have any issue.

**Comment by leedavi** [comment]
This works on my dev system.  Maybe it's something in IIS, or maybe the download corruption.  Does the uploaded zip file open?
